sdk/python/kfp/compiler/_component_builder.py

```diff
@@ -93,90 +93,56 @@ def generate_pip_requirements(self, target_file):
     """ write the python packages to a requirement file
     the generated file follows the order of which the packages are added """
     with open(target_file, 'w') as f:
-      for name, version in self.python_packages.items():
+      for name in sorted(self.python_packages.keys()):
+        version = self.python_packages[name]
         version_str = ''
         if version.has_min_version():
           version_str += ' >= ' + version.min_version + ','
         if version.has_max_version():
           version_str += ' <= ' + version.max_version + ','
         f.write(name + version_str.rstrip(',') + '\n')
 
-class DockerfileHelper(object):
-  """ Dockerfile Helper generates a tarball with dockerfile, ready for docker build
-      arc_dockerfile_name: dockerfile filename that is stored in the tarball """
-
-  def __init__(self, arc_dockerfile_name):
-    self._arc_dockerfile_name = arc_dockerfile_name
-    self._ARC_REQUIREMENT_FILE = 'requirements.txt'
-
-  def _generate_pip_requirement(self, dependency, requirement_filepath):
-    dependency_helper = DependencyHelper()
-    for version in dependency:
-      dependency_helper.add_python_package(version)
-    dependency_helper.generate_pip_requirements(requirement_filepath)
+def _dependency_to_requirements(dependency=[], filename='requirements.txt'):
```

Comment:
nit: dependencies?

Reply:
Yes. Since the exposed interface build_python_component uses the singular, I think it's better to keep it consistent until we upgrade them all together. WDYT?